### ISSUE TITLE
Add Nix support

### ISSUE THREAD

**Issue by Aleksanaa** [MagmaWM/MagmaWM#46]
In case I forgot it. It may depend on these changes
- multi monitor support
- switching config to toml

I plan to use [flake-parts](https://github.com/hercules-ci/flake-parts) instead of [flake-utils](https://github.com/numtide/flake-utils) trash.

**Comment by Aleksanaa** [comment]
An example of converting Nix expressions to toml so that configuration files can be written in Nix:

```nix
{ pkgs, ... }:

let
  tomlFormat = pkgs.formats.toml { };
  toToml = filename: expr: builtins.readFile (tomlFormat.generate filename expr);
in
toToml "config.toml" {
  a = 1;
  b = "foo";
  c = [ "foo" "bar" "baz" ];
  d = {
    foo = bar;
  };
}
```

Then we get this:
```toml
a = 1
b = 2
c = ["foo", "bar", "baz"]

[d]
foo = "bar"
```

The down side is that it's a little bit slower to process.

**Comment by Aleksanaa** [comment]
> I will make it a flake, but we're switching to toml as a config language, so i won't add the nix configuration until the switch is done.

Don't worry, I'll write it when toml is done 🙃

**Comment by electron271** [comment]
dumb question, but dont we already have a nix shell file

**Comment by a-weeb-programmer** [comment]
> dumb question, but dont we already have a nix shell file

Well, I have a flake that contains a dev shell on my local magmawm project directory since I use nixOS and need it to expose project dependencies as build inputs, but I left it to be untracked by git since I don't know if you all wanted it on here. I would add my flake, but Aleksanaa says she is working on a flake, so I'll wait for her to finish it.

**Comment by a-weeb-programmer** [comment]
Btw, I took a look at flake-parts, and it looks like an awesome flake framework for projects like this. Thank you for making me aware that flake-parts exists.

**Comment by electron271** [comment]
due to no updates in a while considering no longer in progress

feel free to add the label if progress is continued

**Comment by electron271** [comment]
resolved by #67

**Comment by nixos-goddess** [comment]
yeah, flake-parts is infinitely better than flake-utils lmao

**Comment by HackedOS** [comment]
This can be closed as #69 is merged

**Comment by electron271** [comment]
nice
